vector/src/main/java/im/vector/app/features/home/room/detail/TimelineFragment.kt

```diff
@@ -1789,87 +1904,81 @@ class RoomDetailFragment @Inject constructor(
         }
     }
 
-//    override fun onFileMessageClicked(eventId: String, messageFileContent: MessageFileContent) {
-//        val isEncrypted = messageFileContent.encryptedFileInfo != null
-//        val action = RoomDetailAction.DownloadOrOpen(eventId, messageFileContent, isEncrypted)
-//        // We need WRITE_EXTERNAL permission
-// //        if (!isEncrypted || checkPermissions(PERMISSIONS_FOR_WRITING_FILES, this, PERMISSION_REQUEST_CODE_DOWNLOAD_FILE)) {
-//            showSnackWithMessage(getString(R.string.downloading_file, messageFileContent.getFileName()))
-//            roomDetailViewModel.handle(action)
-// //        } else {
-// //            roomDetailViewModel.pendingAction = action
-// //        }
-//    }
-
     private fun cleanUpAfterPermissionNotGranted() {
         // Reset all pending data
-        roomDetailViewModel.pendingAction = null
+        timelineViewModel.pendingAction = null
         attachmentsHelper.pendingType = null
     }
 
-//    override fun onAudioMessageClicked(messageAudioContent: MessageAudioContent) {
-//        vectorBaseActivity.notImplemented("open audio file")
-//    }
-
     override fun onLoadMore(direction: Timeline.Direction) {
-        roomDetailViewModel.handle(RoomDetailAction.LoadMoreTimelineEvents(direction))
+        timelineViewModel.handle(RoomDetailAction.LoadMoreTimelineEvents(direction))
     }
 
-    override fun onEventCellClicked(informationData: MessageInformationData, messageContent: Any?, view: View) {
+    override fun onEventCellClicked(informationData: MessageInformationData, messageContent: Any?, view: View, isRootThreadEvent: Boolean) {
         when (messageContent) {
             is MessageVerificationRequestContent -> {
-                roomDetailViewModel.handle(RoomDetailAction.ResumeVerification(informationData.eventId, null))
+                timelineViewModel.handle(RoomDetailAction.ResumeVerification(informationData.eventId, null))
             }
             is MessageWithAttachmentContent      -> {
                 val action = RoomDetailAction.DownloadOrOpen(informationData.eventId, informationData.senderId, messageContent)
-                roomDetailViewModel.handle(action)
+                timelineViewModel.handle(action)
             }
             is EncryptedEventContent             -> {
-                roomDetailViewModel.handle(RoomDetailAction.TapOnFailedToDecrypt(informationData.eventId))
+                if (isRootThreadEvent) {
+                    onThreadSummaryClicked(informationData.eventId, isRootThreadEvent)
+                } else {
+                    timelineViewModel.handle(RoomDetailAction.TapOnFailedToDecrypt(informationData.eventId))
```

Comment:
With this change, there is no way to get UISI details if the event is a root thread event. But when we have a UISI, can the app knows that this is a root thread event? If yes, I think we must change that.
CC @BillCarsonFr

Reply:
No need to investigate that I can remove that to be sure, it will still be clickable the thread summary so we will have both actions :)

Reply:
OK